File: server/src/main/java/au/com/codeka/warworlds/server/store/base/StoreWriter.java
package au.com.codeka.warworlds.server.store.base;

import org.sqlite.javax.SQLiteConnectionPoolDataSource;

import javax.annotation.Nullable;

import au.com.codeka.warworlds.server.store.StoreException;

/** A helper class for writing to the data store. */
public class StoreWriter extends StatementBuilder<StoreWriter> {
  StoreWriter(SQLiteConnectionPoolDataSource dataSource, @Nullable Transaction transaction) {
    super(dataSource, transaction);
  }

  @Override
  public void execute() throws StoreException {
    super.execute();
    if (transaction == null) {
      close();
    }
  }
}

File: server/src/main/java/au/com/codeka/warworlds/server/store/base/StoreWriter.kt
package au.com.codeka.warworlds.server.store.base

import org.sqlite.javax.SQLiteConnectionPoolDataSource

/** A helper class for writing to the data store.  */
class StoreWriter
  internal constructor(dataSource: SQLiteConnectionPoolDataSource, transaction: Transaction?)
  : StatementBuilder<StoreWriter>(dataSource, transaction) {

  override fun execute(): Int {
    val count = super.execute()
    if (transaction == null) {
      close()
    }
    return count
  }
}
